src/main/java/org/nastya/controller/UserController.java

```diff
@@ -35,10 +36,10 @@ public ResponseEntity<UserDTO> getUserByUsername(@PathVariable String username)
     }
 
     @PostMapping
-    public ResponseEntity<UserDTO> createUser(@RequestBody UserDTO userDTO) {
+    public ResponseEntity<UserDTO> createUser(@RequestBody UserDTO userDTO, @RequestHeader(HttpHeaders.AUTHORIZATION) HttpHeaders headers) {
         log.info("Received request to create user: {}", userDTO);
         try {
-            UserDTO savedUser = userService.saveUser(userDTO);
+            UserDTO savedUser = userService.saveUser(userDTO, headers);
```

Comment:
Neverpass anything from controllers that is related to requests to the service layer.